Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs

```diff
@@ -90,7 +91,24 @@ public CosmosClientOptions()
         /// <remarks>
         /// Setting this property after sending any request won't have any effect.
         /// </remarks>
-        public string ApplicationName { get; set; }
+        public string ApplicationName
+        {
+            get => this.applicationName;
+            set
+            {
+                try
+                {
+                    HttpRequestMessage dummyMessage = new HttpRequestMessage(HttpMethod.Head, "https://microsoft.com");
```

Comment:
might be better to just test against localhost instead?

Reply:
You can also try with the empty constructor? https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestmessage.-ctor?view=net-6.0#system-net-http-httprequestmessage-ctor

```suggestion
                    HttpRequestMessage dummyMessage = new HttpRequestMessage();
```